BUILD.plz

```diff
@@ -61,6 +61,171 @@ sh_cmd(
+
+Options:
+  -r, --ref              The ref to create the tag for.
+  -n, --dry-run          Print the commands to be executed.
+  -h, --help             Print this help message.
+
+Tags can automatically be determined for the master branch (next minor)
+and patch branches matching the "*.*.x" pattern (next patch).
+
+For other branches and refs, provide a tag name.
+
+The command automatically detects the current ref,
+but you can also manually choose one using the --ref option.
+When the tag is automatically detected (see above),
+manually provided refs are ignored and are auto-detected."
+
+function info {
+  echo -e "\\\$1" >&2
+}
+
+function warn {
+  echo -e "\x1B[33m\\\$1\x1B[0m" >&2
+}
+
+function error {
+  echo -e "\x1B[31m\\\$1\x1B[0m" >&2
+  exit 1
+}
+
+# Default values of arguments
+REF=""
+DRY_RUN=false
+HELP=false
+ARGS=()
+
+# Loop through arguments and process them
+for arg in "\\\$@"
+do
+    case \\\$arg in
+        -r|--ref)
+        REF="\\\$2"
+        shift 2 # Remove argument name and value from processing
+        ;;
+        -n|--dry-run)
+        DRY_RUN=true
+        shift # Remove --dry-run from processing
+        ;;
+        -h|--help)
+        HELP=true
+        shift # Remove --help from processing
+        ;;
+        *)
+        ARGS+=("\\\$1")
+        shift # Remove generic argument from processing
+        ;;
+    esac
+done
+
+# Setting here so the last shift won't fail the script
+set -o errexit
+
+if [[ \\\$HELP == true ]]; then
+    echo -e "\\\$USAGE"
+
+    exit 0
+fi
+
+# Prepare dry-run command
+CMD=""
+if [[ \\\$DRY_RUN == true ]]; then
+    CMD="echo"
+else
+    CMD=""
+
+    if [[ "\\\$GITHUB_TOKEN" == "" ]]; then
+        error "missing github token"
+    fi
+fi
+
+TAG="\\\${ARGS[0]}"
+
+# Auto-detect tag (if possible)
+if [[ "\\\$TAG" == "" ]]; then
+    if [[ "\\\$REF" != "" ]]; then
+        warn "ignoring ref \\"\\\$REF\\": auto-detecting tag"
+    fi
+
+    CURRENT_BRANCH="\\\$(git branch --show-current)"
+
+    if [[ "\\\$CURRENT_BRANCH" =~ \\\$PATCH_BRANCH_REGEX ]]; then
+        CURRENT_PATCH_VERSION_REGEX="^[vV]?\\\${BASH_REMATCH[1]}\.\\\${BASH_REMATCH[2]}\.(\\\$NAT)$"
+
+        LATEST_VERSION="\\\$(git tag | grep -E "\\\$CURRENT_PATCH_VERSION_REGEX" | sort -r --version-sort | head -1)"
+        if [[ "\\\$LATEST_VERSION" == "" ]]; then
+            error "failed to determine latest version for patch branch \\"\\\$CURRENT_BRANCH\\""
+        fi
+
+        TAG="\\\$($(out_location ///pleasings2//tools/misc:semver) bump patch \\\$LATEST_VERSION)-snapshot.\\\\$(date +%s)"
+    elif [[ "\\\$CURRENT_BRANCH" == "master" ]]; then
+        LATEST_VERSION="\\\$(git tag | grep -E "\\\$VERSION_REGEX" | sort -r --version-sort | head -1)"
+        if [[ "\\\$LATEST_VERSION" == "" ]]; then
+            error "failed to determine latest version"
+        fi
+
+        TAG="\\\$($(out_location ///pleasings2//tools/misc:semver) bump minor \\\$LATEST_VERSION)-snapshot.\\\\$(date +%s)"
```

Comment:
**Note**: post 1.0.0 we could introduce a flag in this command for major version bumped snapshots and conditionally branch on it here.

Reply:
Done